tests/s25Main/integration/testCheats.cpp

```diff
@@ -0,0 +1,62 @@
+// Copyright (C) 2024 Settlers Freaks (sf-team at siedler25.org)
+//
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "Cheats.h"
+#include "worldFixtures/CreateEmptyWorld.h"
+#include "worldFixtures/WorldFixture.h"
+
+BOOST_AUTO_TEST_SUITE(CheatsTests)
+
+namespace {
+template<unsigned T_numPlayers>
+struct CheatsFixture : WorldFixture<CreateEmptyWorld, T_numPlayers>
+{
+    Cheats cheats{this->world};
+};
+using CheatsFixture1P = CheatsFixture<1>;
+using CheatsFixture2P = CheatsFixture<2>;
+} // namespace
+
+BOOST_FIXTURE_TEST_CASE(CheatsAreAllowed_InSinglePlayer, CheatsFixture1P)
+{
+    BOOST_TEST_REQUIRE(cheats.areCheatsAllowed() == true);
+}
+
+BOOST_FIXTURE_TEST_CASE(CheatsAreNotAllowed_InMultiplayer, CheatsFixture2P)
+{
+    BOOST_TEST_REQUIRE(cheats.areCheatsAllowed() == false);
+}
+
+BOOST_FIXTURE_TEST_CASE(CheatModeIsOffByDefault, CheatsFixture1P)
+{
+    BOOST_TEST_REQUIRE(cheats.isCheatModeOn() == false);
+}
+
+BOOST_FIXTURE_TEST_CASE(CanToggleCheatModeOn, CheatsFixture1P)
+{
+    cheats.toggleCheatMode();
+    BOOST_TEST_REQUIRE(cheats.isCheatModeOn() == true);
+}
+
+BOOST_FIXTURE_TEST_CASE(CanToggleCheatModeOnAndOff, CheatsFixture1P)
+{
+    cheats.toggleCheatMode();
+    BOOST_TEST_REQUIRE(cheats.isCheatModeOn() == true);
+    cheats.toggleCheatMode();
+    BOOST_TEST_REQUIRE(cheats.isCheatModeOn() == false);
+}
+
+BOOST_FIXTURE_TEST_CASE(CanToggleCheatModeOnAndOffRepeatedly, CheatsFixture1P)
+{
```

Comment:
Just FTR: IMO it is better to combine such closely related tests, i.e. Just use comments like "initially turned off", "can be turned off" and similar. Especially when using a `WorldFixture` creating the test becomes expensive and if a test fails (which is unlikely for this trivial function anyway) the comment next to the line reported during the failure should be expressive enough.

Especially for CI this can become an issue.   
So just keep that in mind please.
Similar for https://github.com/Return-To-The-Roots/s25client/pull/1688/files#diff-34cfca82833237fac087ed77d47f2465b7e5aff31555b7d6d0c46748d0f93013R33-R68

Not holding off the PR for that though